### DISCUSSION TITLE
Adding  Gcode  before and after probe macro

### DISCUSSION THREAD

**Discussion by Jay-Tech** [terjeio/ioSender#374] [Ideas]
Terjeio,
I was asked if I could write a feature  that would let you add Gcode  before and after  probing ( I guess make one big macro).  I mocked it up in Iosender XL its a dialofg popup that you type the Gcode you want to add before and after, select it to make it activfe, check single use option  or keep it active for the next probe, and  saves it to a file.   I went about this in a few different ways and still testing to see if one is better or not.  One way was when the probe program calls excute I add the addtional gcode to the built probe macro.  Second option was I only add the pre probe Gcode and each probe type has a call back on a successfull probe and the post probe Gcode gets sent out like a macro through the MDI .    

Branch 
https://github.com/Jay-Tech/ioSenderTouch/tree/master
ChangeSet 
https://github.com/Jay-Tech/ioSenderTouch/commit/a05ccfb94d35c79b2b9faed63b51fb5dded978ef
![ScreenCapture](https://github.com/terjeio/ioSender/assets/25882726/9d4e87b2-a656-4680-805a-ec35b69d0b4d)

**Comment by Dietz0r** [comment]
oh yes please! 
Want! :D

**Comment by terjeio** [comment]
I have reworked the code and UI and added this to the latest [edge version](https://www.io-engineering.com/downloads) (p10):

![image](https://github.com/user-attachments/assets/25a0b906-33d3-4741-ad26-8df7d33a97b6)

The drop down functions mainly in the same way as the probing profile does. If a macro is selected the Pre/post macro button is colored and if the mouse pointer is hovered over it the name of the selected is macro is shown. Note that I have not done much testing, I leave it to users to do so and report back if there is any issues.
It can be noted that the post job macro is run on probing job completed event, not added added as a part of the probing "program". One reason for this is that on program completion the edge finders may execute an additional probe in the Z direction, another is that the _Run once_ flag may not be cleared on an error or if the probing job is canceled if part of the program.

Is this ok?

**Reply by Jay-Tech** [reply]
Thank you Terjeio,  looks good to me.   I will let users know about the edge build to get some use and time on it.  Thanks again
